core/services/ocr2/plugins/threshold/decryption_queue.go

```diff
@@ -0,0 +1,208 @@
+package decryption_queue
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+)
+
+type CiphertextId = []byte
+
+type ThresholdDecryptor interface {
+	Decrypt(ctx context.Context, ciphertextId CiphertextId, ciphertext []byte) ([]byte, error)
+}
+
+type DecryptionQueuingService interface {
+	GetRequests(requestCountLimit uint32, totalBytesLimit uint32) []DecryptionRequest
+	GetCiphertext(ciphertextId CiphertextId) ([]byte, error)
+	ResultReady(ciphertextId CiphertextId, plaintext []byte)
+}
+
+type DecryptionRequest struct {
+	ciphertextId CiphertextId
+	ciphertext   []byte
+}
+
+type pendingRequest struct {
+	pendingChan         chan []byte
+	ciphertext          []byte
+	expirationTimestamp uint64
```

Comment:
expirationTimestamp fields are unused in both structs.

Reply:
I can either remove this or keep for logging purposes.

Reply:
I'd remove. Speaking of logging - consider passing a Logger object to NewThresholdDecryptor() and use it for logs (debug-level?) across the code.